tests/test_pipeline_mixin.py

```diff
@@ -236,67 +307,89 @@ def run_pipeline_test(
+        except Exception:
+            logger.warning(
+                f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')}_{torch_dtype} is skipped: Could not find or load "
+                f"the model from `{repo_id}` with `{model_architecture}`."
+            )
+            self.skipTest(f"Could not find or load the model from {repo_id} with {model_architecture}.")
+
+        # -------------------- Load tokenizer --------------------
+
         tokenizer = None
         if tokenizer_name is not None:
-            tokenizer_class = getattr(transformers_module, tokenizer_name)
-            tokenizer = tokenizer_class.from_pretrained(repo_id, revision=commit)
-
-        processor = None
-        if processor_name is not None:
-            processor_class = getattr(transformers_module, processor_name)
-            # If the required packages (like `Pillow` or `torchaudio`) are not installed, this will fail.
             try:
-                processor = processor_class.from_pretrained(repo_id, revision=commit)
+                tokenizer_class = getattr(transformers_module, tokenizer_name)
+                tokenizer = tokenizer_class.from_pretrained(repo_id, revision=commit)
             except Exception:
                 logger.warning(
-                    f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')}_{torch_dtype} is skipped: Could not load the "
-                    f"processor from `{repo_id}` with `{processor_name}`."
+                    f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')}_{torch_dtype} is skipped: "
+                    f"Could not load the tokenizer from `{repo_id}` with `{tokenizer_name}`."
                 )
-                self.skipTest(f"Could not load the processor from {repo_id} with {processor_name}.")
+                self.skipTest(f"Could not load the tokenizer from {repo_id} with {tokenizer_name}.")
+
+        # -------------------- Load processors --------------------
+
+        processors = {}
+        for key, name in zip(
+            ["image_processor", "feature_extractor", "processor"],
+            [image_processor_name, feature_extractor_name, processor_name],
+        ):
+            if name is not None:
+                try:
+                    # Can fail if some extra dependencies are not installed
+                    processor_class = getattr(transformers_module, name)
+                    processor = processor_class.from_pretrained(repo_id, revision=commit)
+                    processors[key] = processor
+                except Exception:
+                    logger.warning(
+                        f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')}_{torch_dtype} is skipped: "
+                        f"Could not load the {key} from `{repo_id}` with `{name}`."
+                    )
+                    self.skipTest(f"Could not load the {key} from {repo_id} with {name}.")
```

Comment:
It is designed the same way on the `main` now, according to the comment some instances might not be loaded due to optional dependencies
https://github.com/huggingface/transformers/blob/af638c4afea81ac251f1ae63497789be79e0baaa/tests/test_pipeline_mixin.py#L256-L267

Reply:
Currently on `main`, this `run_pipeline_test` only take a single `tokenizer_name` and a **single** `processor_name`, and try to load them. So if a specified `processor_name` could not be loaded, we can skip it.

However, here in this PR, we try to **collect** the processors and put them into the dictionary `processors`. So logically it's better not to skip the test during this collection.

(But in practice, this may not produce any difference)